src/core/src/op/result.cpp

```diff
@@ -31,52 +28,55 @@ void op::v0::Result::validate_and_infer_types() {
     output.set_tensor_ptr(input.get_tensor_ptr());
 }
 
-std::shared_ptr<Node> op::v0::Result::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> Result::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v0_Result_clone_with_new_inputs);
     check_new_args_count(this, new_args);
 
-    auto res = std::make_shared<Result>(new_args.at(0));
-    return std::move(res);
+    return std::make_shared<Result>(new_args.at(0));
 }
 
-bool op::v0::Result::evaluate(ov::TensorVector& outputs, const ov::TensorVector& inputs) const {
+bool Result::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v0_Result_evaluate);
     OPENVINO_ASSERT(inputs.size() == 1);
-    if (outputs.empty())
-        outputs.emplace_back(ov::Tensor(inputs[0].get_element_type(), inputs[0].get_shape()));
-    else
+
+    if (outputs.empty()) {
+        outputs.emplace_back(inputs[0].get_element_type(), inputs[0].get_shape());
+    } else if (!outputs[0]) {
```

Comment:
The change isn't equivalent for previous behavior - number of outputs will be checked only when no tensor created for 0th output

Reply:
I can restore it. But here at least one tensor is in vector and do assertion is little redundant, as for data copy only first one is used.